lib/translator/maintenance_in.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "maintenance_in.h"
+
+#include "lib/scsi.h"
+
+namespace translator {
+
+StatusCode ValidateReportSupportedOpCodes(Span<const uint8_t> scsi_cmd) {
+  scsi::ReportOpCodesCommand report_cmd = {};
+  if (!ReadValue(scsi_cmd, report_cmd)) {
+    DebugLog("Malformed Report Supported OpCodes command");
+    return StatusCode::kInvalidInput;
+  }
+
+  if (report_cmd.requested_op_code !=
+          static_cast<uint8_t>(scsi::OpCode::kWriteSame16) ||
+      report_cmd.reporting_options != 0b001) {
+    DebugLog("Only supporting ReportSupportedOpCodes for WriteSame16");
```

Comment:
Why do we only support this one niche case?

Reply:
This was in the project specs: " only report if we support “WriteSame16”. I think it might be because on start up/connection the scsi device queries whether WriteSame16 is supported?

Reply:
good to add a comment about this

Reply:
there's a comment in the header file describing this but it definitely makes a lot of sense to put a comment here